ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/facade/DefaultAuditor.java

```diff
@@ -48,10 +58,38 @@ public void setup()
     @Override
     public void audit(AuditEntry logEntry)
     {
-        if (!filter.isFiltered(logEntry))
+        if (shouldAudit(logEntry))
+        {
+            performAudit(logEntry);
+        }
+    }
+
+    private boolean shouldAudit(AuditEntry logEntry)
+    {
+        long start = System.nanoTime();
+        try
+        {
+            return !filter.isFiltered(logEntry);
+        }
+        finally
+        {
+            long end = System.nanoTime();
+            auditMetrics.filterAuditRequest(end - start, TimeUnit.NANOSECONDS);
+        }
+    }
+
+    private void performAudit(AuditEntry logEntry)
```

Comment:
IMO it would make sense to perform the obfuscation step before starting the timer? That would make the metric value more "true".

Reply:
Sure, I've moved the obfuscation outside of the performAudit method